### PR TITLE
Fix/cm-504 .net test fixes

### PR DESCRIPTION
**Pull request by mgcouchbase**
.net only fix: tested the test cases locally

### CODE REVIEW

File: CBLClient/Apps/CBLTestServer-Dotnet/TestServer/ValueSerializer.cs [hunk 1]
```diff
@@ -117,7 +125,8 @@ internal static object DeserializeValue(String value)
             }
             else if (value.StartsWith("\"") && value.EndsWith("\""))
             {
-                return value.Replace("\"", String.Empty);
+                string trimVal = value.Substring(1, value.Length - 2);
+                return trimVal;
```
Comment (euniceatcouchbase) [inline CBLClient/Apps/CBLTestServer-Dotnet/TestServer/ValueSerializer.cs:129]: do you want to 
`return value.Stustring(1, value.length - 2)`
?

Comment (mgcouchbase) [reply CBLClient/Apps/CBLTestServer-Dotnet/TestServer/ValueSerializer.cs:129]: In java and other languages we are using  return (T) value.substring(1, value.length() - 1);, But in dot net  -2  is exclusive Ex:  String a = "g=nhz";
		Console.WriteLine(a.Substring(1, a.Length-2));    OUTPUT  =nh

Comment (euniceatcouchbase) [reply CBLClient/Apps/CBLTestServer-Dotnet/TestServer/ValueSerializer.cs:129]: i mean to combine the 2 lines into 1, trimVal seems not needed. 
I am not questioning the index value

Comment (mgcouchbase) [reply CBLClient/Apps/CBLTestServer-Dotnet/TestServer/ValueSerializer.cs:129]: If string is " a"lkl"l " previous  code returns alkll, where as current code returns a"lk"l

Comment (sridevi-15) [reply CBLClient/Apps/CBLTestServer-Dotnet/TestServer/ValueSerializer.cs:129]: I think Eunice meant to combine line #128 and #129. She is referring to new code, not old code. new code is fine, you return the value directly as it is one line.

File: CBLClient/Apps/CBLTestServer-Dotnet/TestServer/ValueSerializer.cs [hunk 2]
```diff
@@ -38,9 +38,15 @@ public static string Serialize(Object value, Type t)
             {
                 return "L" + value;
             }
-            else if (t.Equals(typeof(float)) || t.Equals(typeof(double)))
+            else if (t.Equals(typeof(float)))
```
Comment (sridevi-15) [inline CBLClient/Apps/CBLTestServer-Dotnet/TestServer/ValueSerializer.cs:41]: this is required only in .Net to have differrent if condition of Float and double. Other platforms we combine. Hope Jim confirmed this.

**Review by sridevi-15 — APPROVED** [review]

**Review by euniceatcouchbase — APPROVED** [review]

File: CBLClient/Apps/CBLTestServer-Dotnet/TestServer/ValueSerializer.cs (hunk 2 above)
Comment (mgcouchbase) [reply CBLClient/Apps/CBLTestServer-Dotnet/TestServer/ValueSerializer.cs:41]: Yes other platforms have different conditions

**Pull request merged**